src/app.rs

```diff
@@ -2813,6 +2813,58 @@ mod test {
         }
     }
 
+    mod contract_instantiation {
+        #[test]
+        #[cfg(feature = "cosmwasm_1_2")]
+        fn instantiate2_works() {
+            use super::*;
+            use cw_utils::*;
+
+            // prepare application and actors
+            let mut app = App::default();
+            let sender = Addr::unchecked("sender");
+
+            // store contract's code
+            #[cfg(not(feature = "multitest_api_1_0"))]
+            let code_id = app.store_code_with_creator(Addr::unchecked("creator"), echo::contract());
+            #[cfg(feature = "multitest_api_1_0")]
+            let code_id = app.store_code(Addr::unchecked("creator"), echo::contract());
+
+            // initialize the contract
+            let init_msg = to_binary(&Empty {}).unwrap();
+            let msg = WasmMsg::Instantiate2 {
+                admin: None,
+                code_id,
+                msg: init_msg,
+                funds: vec![],
+                label: "label".into(),
+                salt: [0, 1, 2, 3, 4, 5].as_slice().into(),
+            };
+            let res = app.execute(sender, msg.into()).unwrap();
+
+            // assert a proper instantiate result
+            let parsed = parse_instantiate_response_data(res.data.unwrap().as_slice()).unwrap();
+            assert!(parsed.data.is_none());
+
+            // assert contract's address is exactly the predicted one
+            //
+            // REMARK:
```

Comment:
It would be nice if we could add this somewhere where the user can actually see it, but I guess there is no good place for this currently. Maybe a section in the readme or somewhere in the api docs where we collect these would be a good idea? (not in this PR, just a general thing I noticed)

Reply:
I was thinking about something like Mutli-Test Book just like Bart and Jan did it for casmwasm and Sylvia, but I guess now it is challenging for me, I mean I know to little yet to provide highest quality documentation for the whole library. I guess changelog mey be a good place.